Automattic-Tracks-iOS/Event Logging/LogEncryptor.swift

```diff
@@ -0,0 +1,104 @@
+import Foundation
+import Sodium
+
+class LogEncryptor {
+
+    private let chunkSize = 4096
+    private let publicKey: Bytes
+
+    enum LogEncryptorError: Error {
+        case unableToReadFile
+        case unableToEncryptFile
+        case logSecretTooLong
+        case unableToWriteFile
+    }
+
+    init(withPublicKey key: Bytes) {
+        self.publicKey = key
+    }
+
+    func encryptLog(_ log: LogFile) throws -> URL {
+
+        /// Set up our output path
+        let uniqueSegment = UUID().uuidString + "-" + log.url.lastPathComponent
+        let outputURL = URL(fileURLWithPath: NSTemporaryDirectory()).appendingPathComponent(uniqueSegment)
+        assert(outputURL.path != log.url.path, "The output file path must be unique")
+
+        /// Do the encrypted stream setup
+        let sodium = Sodium()
+        let secretkey = sodium.secretStream.xchacha20poly1305.key()
+        let encryptedKey = try encryptSecretWithSodium(secret: secretkey)
+        let stream_enc = sodium.secretStream.xchacha20poly1305.initPush(secretKey: secretkey)!
+        let header = stream_enc.header()
+
+        /// Prep our file handles
+        try initializeOutputFile(at: outputURL)
+        let inputFileHandle = try FileHandle(forReadingFrom: log.url)
+        let outputFileHandle = try FileHandle(forWritingTo: outputURL)
+
+        defer {
+            inputFileHandle.closeFile()
+            outputFileHandle.closeFile()
+        }
+
+        /// Write JSON Preamble
+        outputFileHandle.write("""
+        {
+            "keyedWith": "v1",
```

Comment:
I thought about that, but in this case, because it's a file format, that's a completely immutable value – it must never change. As you suggest, using a constant for this and referencing it in two places would make it too easy to change.

After chatting with @oguzkocer, I've added some inline documentation in 72225dc5ac7b182ac7d061d24714e40f3206813c to explain this.